frontend/src/components/cronjob/Details.tsx

```diff
@@ -31,11 +60,74 @@ export default function CronJobDetails() {
 
   const ownedJobs = filterOwnedJobs(jobs);
 
+  function handleCron(suspend: boolean) {
+    const suspendingOrStarting = suspend ? 'Suspend' : 'Resum';
+    cronJob.spec.suspend = suspend;
+    // when we apply the cronjob, we need to remove the resourceVersion because a new resourceVersion is generated
+    // when the cronjob is applied
+    delete cronJob.metadata.resourceVersion;
+    setIsCheckingCronSuspendStatus(true);
+    dispatch(
+      clusterAction(() => applyFunc(cronJob.jsonData), {
+        startMessage: t(`resource|${suspendingOrStarting}ing CronJob {{ newItemName }}…`, {
+          newItemName: cronJob.metadata.name,
+        }),
+        cancelledMessage: t(
+          `resource|Cancelled ${suspendingOrStarting}ing CronJob {{ newItemName }}.`,
+          {
+            newItemName: cronJob.metadata.name,
+          }
+        ),
+        successMessage: t(`resource|${suspendingOrStarting}ed CronJob {{ newItemName }}.`, {
+          newItemName: cronJob.metadata.name,
+        }),
+        errorMessage: t(
+          `resource|Failed to ${suspend ? 'suspend' : 'resume'} CronJob {{ newItemName }}.`,
+          {
+            newItemName: cronJob.metadata.name,
+          }
+        ),
+      })
+    );
+  }
+
+  const actions = [];
+
+  if (isCronSuspended) {
+    actions.push(
+      <ActionButton
+        description={t('frequent|Play')}
+        onClick={() => {
+          handleCron(false);
+        }}
+        icon="mdi:play-circle"
+        iconButtonProps={{
+          disabled: isCheckingCronSuspendStatus,
+        }}
+      />
+    );
+  } else {
+    actions.push(
+      <ActionButton
+        description={t('frequent|Pause')}
+        onClick={() => {
+          handleCron(true);
+        }}
+        icon="mdi:pause-circle"
+        iconButtonProps={{
+          disabled: isCheckingCronSuspendStatus,
+        }}
+      />
+    );
```

Comment:
I'd probably have just one button change its label/icons/function depending on what state we have.

Reply:
fixed